integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXEncryptedVolumeTest.java

```diff
@@ -278,6 +281,9 @@ private EnvironmentTestDto verifyEnvironmentResponseDiskEncryptionKey(TestContex
         } else if (CloudPlatform.AZURE.name().equals(environment.getCloudPlatform())) {
             String diskEncryptionSetId = environment.getAzure().getResourceEncryptionParameters().getDiskEncryptionSetId();
             verifyDiskEncryptionKey(testDto.getCloudPlatform(), diskEncryptionSetId, testDto.getRequest().getName());
+        } else if (CloudPlatform.GCP.name().equals(environment.getCloudPlatform())) {
+            String encryptionKey = environment.getGcp().getGcpResourceEncryptionParameters().getEncryptionKey();
+            verifyDiskEncryptionKey(testDto.getCloudPlatform(), encryptionKey, testDto.getRequest().getName());
```

Comment:
this class is full with `if` and `else if` conditions based on cloud provider. you should use `CloudFunctionality` interface instead and implement proper methods/assertions if possible for every provider

Reply:
Updated.

Reply:
Please move these to the [CloudFunctionality](https://github.com/hortonworks/cloudbreak/blob/master/integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/CloudFunctionality.java). There is already `CloudFunctionality.listVolumeEncryptionKeyIds` that is part of the already existing provider utils with the `CloudFunctionality`.

Reply:
[CloudFunctionality](https://github.com/hortonworks/cloudbreak/blob/master/integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/CloudFunctionality.java) is used to make the cloud specific calls [CloudProvider](https://github.com/hortonworks/cloudbreak/blob/master/integration-test/src/main/java/com/sequenceiq/it/cloudbreak/cloud/v4/CloudProvider.java) on the other hand integrates CDP with specific cloud. These checks are CDP-per cloud specific, so I have put them in CloudProvider. There are other similar methods in AWS/Azure CloudProviders.

Please let me know if there are still concerns.

Reply:
You were right, we cannot move the entire validation to the [CloudFunctionality](https://github.com/hortonworks/cloudbreak/blob/master/integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/CloudFunctionality.java). So I've made a refactor on these methods and fixed the `verifyDiskEncryptionKey` and `verifyVolumeEncryptionKey` calls in `CloudProviderProxy`.